### PR TITLE
dns_utils: use fallback if the default resolver does not support DNSSEC

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/common/dns_utils.cpp [hunk 1]
```diff
@@ -266,11 +277,27 @@ DNSResolver::DNSResolver() : m_data(new DNSResolverData())
...
+    bool available, valid;
+    static const char *probe_hostname = "updates.moneropulse.org";
+    auto records = get_txt_record(probe_hostname, available, valid);
+    if (!valid)
+    {
+      MINFO("Failed to verify DNSSEC record from " << probe_hostname << ", falling back to well known DNSSEC resolvers");
+      ub_ctx_delete(m_data->m_ub_context);
+      m_data->m_ub_context = ub_ctx_create();
+      add_anchors(m_data->m_ub_context);
+      for (const auto &ip: dns_public_addr)
```
Comment (vtnerd) [inline src/common/dns_utils.cpp]: Isn't this guaranteed to be empty (`!DNS_PUBLIC` implies that `dns_public_addr` is empty)? This doesn't seem to do anything, and not what the comments indicate. Am I missing something?

Comment (moneromooo-monero) [reply src/common/dns_utils.cpp]: Good point. I tended to use the global array of well known servers. This implies that it's just the switch to tcp that fixes it, with no need for extra servers... I'll change it to do just that then.

Comment (moneromooo-monero) [reply src/common/dns_utils.cpp]: Actually, that is not enough. The resolvconf/hosts calls also need to be not made, at least on my setup.

Comment (moneromooo-monero) [reply src/common/dns_utils.cpp]: I just added the default servers. This seems like the more conservative option.

**Pull request merged**